py/server/deephaven/pandas.py

```diff
@@ -22,12 +23,26 @@
 _is_dtype_backend_supported = pd.__version__ >= "2.0.0"
 
 
-def _column_to_series(table: Table, col_def: Column) -> pd.Series:
+_DTYPE_NULL_MAPPING: Dict[DType, Tuple] = {
+    dtypes.bool_: (NULL_BOOLEAN, pd.BooleanDtype),
+    dtypes.byte:  (NULL_BYTE, pd.Int8Dtype),
+    dtypes.short: (NULL_SHORT, pd.Int16Dtype),
+    dtypes.char: (NULL_CHAR, pd.UInt16Dtype),
+    dtypes.int32: (NULL_INT, pd.Int32Dtype),
+    dtypes.int64: (NULL_LONG, pd.Int64Dtype),
+    dtypes.float32: (NULL_FLOAT, pd.Float32Dtype),
+    dtypes.float64: (NULL_DOUBLE, pd.Float64Dtype),
+}
+
+
+def _column_to_series(table: Table, col_def: Column, conv_null: bool = False) -> pd.Series:
     """Produce a copy of the specified column as a pandas.Series object.
 
     Args:
         table (Table): the table
         col_def (Column):  the column definition
+        conv_null (bool): whether to check for Deephaven nulls in the data and automatically replace them with
+            pd.NA, default is False.
```

Comment:
I am not sure if we need to offer 'null-to-NaN' conversion:
1. pandas nullable extension types don't distinguish between pd.NA and NaN. 
2. Converting integer nulls to NaNs results in auto promoting the numpy arrays to np.float64, which could be problematic 
3. Converting floating nulls to NaNs will break round-tripping between DH and Pandas

Reply:
I think just going with `pd.NA` is probably best, but I want to get feedback from @alexpeters1208 and @jjbrosnan

Reply:
If `pd.NA` is what users would expect, I would go with that. If I convert other stuff from numpy or a list or something, do nulls get converted to `pd.NA`? If the expectation when Pandas sees `null` is to do nothing, then I would do that.

Reply:
`pd.NA` was what I expected when I used Pandas more often. That was 2+ years ago, so that could be outdated now, but I don't see a good reason for them to change that.